activity-tracker-root/activity-tracker/src/main/java/io/fourfinance/activity_tracker/activity/TrackUserActivityAspect.java

```diff
@@ -0,0 +1,57 @@
+package io.fourfinance.activity_tracker.activity;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.annotation.AfterReturning;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.aspectj.lang.reflect.MethodSignature;
+
+import io.fourfinance.activity_tracker.audit.TrackUserActivityAudits;
+
+@Aspect
+public class TrackUserActivityAspect {
+
+    public static final String ACTIVITY_NAME = "activity";
+    private final ActivityParameters activityParameters;
+    private final TrackUserActivityAudits trackUserActivityAudits;
+    private final TrackUserActivityMetrics trackTrackUserActivityMetrics;
+
+    public TrackUserActivityAspect(TrackUserActivityAudits trackUserActivityAudits, TrackUserActivityMetrics trackTrackUserActivityMetrics,
+                                   ActivityParameters activityParameters) {
+        this.trackUserActivityAudits = trackUserActivityAudits;
+        this.trackTrackUserActivityMetrics = trackTrackUserActivityMetrics;
+        this.activityParameters = activityParameters;
+    }
+
+    @Before("@annotation(io.fourfinance.activity_tracker.activity.TrackUserActivity)")
+    public void audit(JoinPoint joinPoint) {
+        Map<String, String> values = new HashMap<>();
+        values.put(ACTIVITY_NAME, readActivityName(joinPoint));
+        for (String key : activityParameters.getParameters()) {
+            values.put(key, extractParamValue(joinPoint, key));
+        }
+        trackUserActivityAudits.audit(values);
+    }
+
+    @AfterReturning("@annotation(io.fourfinance.activity_tracker.activity.TrackUserActivity)")
+    void processMetrics(JoinPoint joinPoint) {
+        String activityName = readActivityName(joinPoint);
+        trackTrackUserActivityMetrics.process(activityName);
+    }
+
+    private String readActivityName(JoinPoint joinPoint) {
+        MethodSignature signature = (MethodSignature) joinPoint.getSignature();
+        TrackUserActivity annotation = signature.getMethod().getAnnotation(TrackUserActivity.class);
+        if (annotation != null) {
```

Comment:
It's possible to have method signature joint point with something annotated with @TrackUserActivity ? It looks like you only gave possibility to annotate methods so if you have annotations like this(@annotation) on this aspect you never get possibility to have method not annotated.